stage0_bin/src/asm/boot.s

```diff
@@ -43,7 +43,10 @@ gp_handler:
 vc_handler:
     pop %ebx              # get the error code
     cmp $0x72, %ebx       # is this about CPUID?
-    jne 2f                # if not, skip ahead
+    jne 2f                # if not, skip ahead and crash
+    pop %ebx              # get the instruction pointer
```

Comment:
This pops the EIP value off the stack, I think that'll break the `IRET` later (where should it return to?).

What you'd really want to do is effectively `((%esp))` (first dereference ESP to get the value of EIP, then dereference EIP) but I'm pretty sure you can't do it like that.

Reply:
Ah yes, good point.